include/cantera/oneD/refine.h

```diff
@@ -69,51 +123,142 @@ class Domain1D;
         return m_gridmin;
     }
 
-    //! Determine locations in the grid that need additional grid points.
-    //!
-    //! @returns The number of new grid points needed (size of #m_loc)
+    /**
+     * Determine locations in the grid that need additional grid points and
+     * updates the internal state of the Refiner with this information.
+     *
+     * @param[in] n  Number of grid points
+     * @param[in] z  Point to array of grid points
+     * @param[in] x  Pointer to solution vector
+     *
+     * @returns The number of new grid points needed (size of #m_insertPts)
+     */
     int analyze(size_t n, const double* z, const double* x);
+
+    /**
+     * Constructs a new grid based on refinement locations determined by the analyze
+     * method.
+     *
+     * This function generates a new grid by inserting additional points into the
+     * current grid at locations where the `analyze` method has identified a need for
```

Comment:
This will generate a Doxygen link to the member function in the same class:
```suggestion
     * current grid at locations where the analyze() method has identified a need for
```